opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -675,8 +659,45 @@ boolean hasFileBasedTags() {
         return true;
     }
 
+    /**
+     * @param dotGit {@code .git} file
+     * @return value of the {@code gitdir} property from the file
+     */
+    private String getGitDirValue(File dotGit) {
+        try (Scanner scanner = new Scanner(dotGit, StandardCharsets.UTF_8)) {
```

Comment:
Nothing wrong with this but maybe the functional approach is more readable?
```java
Files.lines(...)
    .filter(line -> line.startsWith(Constants.GITDIR))
    .map(line -> line.substring(Constants.GITDIR.length()))
    .findFirst()
    .orElse(null);
```

Reply:
depends on personal preference I'd say. To me the classic way looks more readable.